src/slotshandler.cpp

```diff
@@ -103,9 +103,16 @@ int file_size (int file_num)
    }
 }
 
-#define FILL_SLOT_INFO(filevar, processedvar, file) \
-    (filevar) = (file); \
-    (processedvar) = true;
+inline bool fillSlot(std::string &filevar, bool &processedvar, const std::string fullpath, std::string extension, const std::string type_ext, const std::string type_desc) {
```

Comment:
To avoid a copy, you should prefer passing the strings by reference (e.g const std::string& extension)